pydantic/_internal/_dataclasses.py

```diff
@@ -170,7 +170,7 @@ def __init__(__dataclass_self__: PydanticDataclass, *args: Any, **kwargs: Any) -
 
     cls.__pydantic_core_schema__ = schema
     cls.__pydantic_validator__ = validator = create_schema_validator(
-        schema, core_config, config_wrapper.plugin_settings
+        schema, f'{cls.__module__}:{cls.__qualname__}', core_config, config_wrapper.plugin_settings
```

Comment:
I think it would be useful to have a function in the plugin code:

```py
def build_type_path(module: str, name: str) -> str:
    return f'{module}:{name}'
```